app/middlewares/rack/current_locale.rb

```diff
@@ -0,0 +1,86 @@
+module Rack
+  class CurrentLocale
+    def initialize app
+      @app = app
+    end
+
+    def call env
+      @req = Rack::Request.new(env)
+
+      set_current_locale
+      set_current_locale_from_subdomain
+      redirect_to_locale_subdomain
+
+      @app.call(env)
+    end
+
+    private
+
+    def set_current_locale
+      Current.locale = ::Locale.find_by abbreviation: I18n.locale
+    end
+
```

Comment:
this is why the tests are failing.

You are making it so that the routing logic now depends on the locale database tables to exist an be populated

this will be `nil` in the specs because there are no locales, and that will make line 36 fail.

you can get the specs passing by putting `before { create(:locale, :en) }` at the top of the failing specs, but this brings up a question for me

should middleware routing be tied to the state of the DB? and does making a database call outside of application code introduce any risks (security or otherwise)

I am not familiar enough with rack to really know, but if it's fine then adding the `before` in failing specs should do the trick